src/io/memory.cr

```diff
@@ -393,7 +393,11 @@ class IO::Memory < IO
   # ```
   def to_s : String
     if encoding = @encoding
-      String.new to_slice, encoding: encoding.name, invalid: encoding.invalid
+      {% if flag?(:without_iconv) %}
+        raise NotImplementedError.new("String.encode")
```

Comment:
Wouldn't be better to move this check into the `String.encode` method instead?

Reply:
Yeah, I wondered about that. But `String.encode` is explicit about being for encoding purposes. Calling that method with `-Dwithout_iconv` should lead to a compile time error. `IO::Memory#to_s` on the other hand is not explicitly about encoding. It's just something that it needs to do in some state. In this case there should definitely not be a compile time error. So raising the runtime error here makes more sense IMO.